components/suggest/src/fakespot.rs

```diff
@@ -0,0 +1,225 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/// Fakespot-specific logic
+
+/// Score used to order Fakespot suggestions
+///
+/// FakespotScore contains several components, each in the range of [0, 1]
+pub struct FakespotScore {
+    /// Did the query match the `keywords` field exactly?
+    keywords_score: f64,
+    /// How well did the query match the `product_type` field?
+    product_type_score: f64,
+    /// Fakespot score from the RS data, this reflects the average review, number of reviews,
+    /// Fakespot grade, etc.
+    fakespot_score: f64,
+}
```

Comment:
Splitting this out into it's own struct isn't needed, but I felt like it helped documenting the details of each component.